### PR TITLE
Feature: Add animations to foldable section

### PR DESCRIPTION
**Pull request by johnfewell**
# Description

Refactored the foldable section. The design is inspired by [Material Design's expansion panel](https://m1.material.io/components/expansion-panels.html#). The existing design has the chevron alternating from pointing down to pointing right. This is misleading because there is no subordinate information to the right. Also, this PR animates the arrow and foldable section, creating a delightful experience.

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

## Screenshots (if appropriate):
[expansion.webm](https://github.com/QuivrHQ/quivr/assets/1273463/71ef2c98-c0ed-4374-840e-0bf16bf4da55)
### Update
[expansion2.webm](https://github.com/QuivrHQ/quivr/assets/1273463/0be769d2-93b4-42e3-938d-3bc0a63e0e06)

### CODE REVIEW

**Review by Zewed — CHANGES_REQUESTED** [review]

**Comment by Zewed** [comment]
Nice work :) 
Just need some small changes to be merged :)

File: frontend/lib/components/ui/FoldableSection/FoldableSection.tsx [hunk 1]
```diff
@@ -35,12 +35,21 @@ export const FoldableSection = (props: FoldableSectionProps): JSX.Element => {
           <p className={styles.header_title}>{props.label}</p>
         </div>
         <Icon
-          name={folded ? "chevronDown" : "chevronRight"}
+          name="chevronDown"
           size="normal"
           color="black"
+          classname={`${styles.iconRotate} ${
+            folded ? styles.iconRotateDown : styles.iconRotateUp
+          }`}
```
[the remark this answers is not in the document]
Comment (johnfewell) [reply frontend/lib/components/ui/FoldableSection/FoldableSection.tsx]: Done.

File: frontend/lib/components/ui/FoldableSection/FoldableSection.module.scss [hunk 2]
```diff
@@ -11,6 +11,19 @@
   overflow: hidden;
   font-size: Typography.$small;
 
+  .contentWrapper {
+    overflow: hidden;
+    transition: max-height 0.3s cubic-bezier(0.65, 0.05, 0.36, 1);
+  }
+
+  .contentExpanded {
+  max-height: 200px;
```
[the remark this answers is not in the document]
Comment (johnfewell) [reply frontend/lib/components/ui/FoldableSection/FoldableSection.module.scss]: Done.

File: frontend/lib/components/ui/FoldableSection/FoldableSection.module.scss [hunk 3]
```diff
@@ -11,6 +11,19 @@
   overflow: hidden;
   font-size: Typography.$small;
 
+  .contentWrapper {
+    overflow: hidden;
+    transition: max-height 0.3s cubic-bezier(0.65, 0.05, 0.36, 1);
```
[the remark this answers is not in the document]
Comment (johnfewell) [reply frontend/lib/components/ui/FoldableSection/FoldableSection.module.scss]: Done.

**Comment by johnfewell** [comment]
@Zewed Thank you for your feedback; the issues have been addressed.

**Review by Zewed — APPROVED** [review]

**Pull request merged**